From b844470ae8ce54bf102329cd024eed9ae50bc5c3 Mon Sep 17 00:00:00 2001
From: Phantom X <PhantomX@users.noreply.github.com>
Date: Thu, 4 Feb 2021 15:26:30 -0300
Subject: [PATCH] Revert microVU: Implement Overflow checks. Fixes Superman
 Returns  Removed patches for Superman Returns

Revert commit ee07f860fc1dd7e8ada6c13903bb3a592b8d4864, this breaks older CPUs
---
 bin/GameIndex.yaml          | 34 ++++++++++++++++++++++++++++++++++
 pcsx2/x86/microVU_Upper.inl | 19 -------------------
 2 files changed, 34 insertions(+), 19 deletions(-)

diff --git a/bin/GameIndex.yaml b/bin/GameIndex.yaml
index dc05e9f..580778e 100644
--- a/bin/GameIndex.yaml
+++ b/bin/GameIndex.yaml
@@ -13412,6 +13412,15 @@ SLES-53744:
 SLES-53746:
   name: "Superman Returns"
   region: "PAL-E"
+  patches:
+    E8F7BAB6:
+      content: |-
+        author=kozarovv
+        // Fix sps and various graphical issues by Using iaddiu instead of FSAND
+        // require preload frame data and crc hack atleast minimum for GSDX HW
+        patch=1,EE,00639Ef0,word,10050208
+        patch=1,EE,0063a068,word,10080208
+        patch=1,EE,0063D488,word,10020208
 SLES-53747:
   name: "Ed, Edd, 'n Eddy - The Misadventure"
   region: "PAL-E"
@@ -14585,12 +14594,28 @@ SLES-54348:
 SLES-54349:
   name: "Superman Returns"
   region: "PAL-I"
+  patches:
+    E7F7B6BD:
+      content: |-
+        author=kozarovv
+        // Use iaddiu instead of FSAND.
+        patch=1,EE,00639ef0,word,10050208
+        patch=1,EE,0063a068,word,10080208
 SLES-54350:
   name: "Superman Returns"
   region: "PAL-G"
 SLES-54351:
   name: "Superman Returns"
   region: "PAL-S"
+  patches:
+    EDF0A0A7:
+      content: |-
+        author=kozarovv
+        // Fix sps and various graphical issues by Using iaddiu instead of FSAND
+        // require preload frame data and crc hack atleast minimum for GSDX HW
+        patch=1,EE,00639Ef0,word,10050208
+        patch=1,EE,0063a068,word,10080208
+        patch=1,EE,0063D488,word,10020208
 SLES-54354:
   name: "Final Fantasy XII"
   region: "PAL-E"
@@ -36999,6 +37024,15 @@ SLUS-21434:
   name: "Superman Returns - The Video Game"
   region: "NTSC-U"
   compat: 4
+  patches:
+    E1BF5DCA:
+      content: |-
+        author=kozarovv
+        // Fix sps and various graphical issues by Using iaddiu instead of FSAND
+        // require preload frame data and crc hack atleast minimum for GSDX HW
+        patch=1,EE,00639E70,word,10050208
+        patch=1,EE,00639FE8,word,10080208
+        patch=1,EE,0063D408,word,10020208
 SLUS-21435:
   name: "One Piece - Grand Adventure"
   region: "NTSC-U"
diff --git a/pcsx2/x86/microVU_Upper.inl b/pcsx2/x86/microVU_Upper.inl
index 14523ff..9b378bf 100644
--- a/pcsx2/x86/microVU_Upper.inl
+++ b/pcsx2/x86/microVU_Upper.inl
@@ -23,12 +23,6 @@
 #define ADD_XYZW			((_XYZW_SS && modXYZW) ? (_X ? 3 : (_Y ? 2 : (_Z ? 1 : 0))) : 0)
 #define SHIFT_XYZW(gprReg)	{ if (_XYZW_SS && modXYZW && !_W) { xSHL(gprReg, ADD_XYZW); } }
 
-
-const __aligned16 u32 sse4_compvals[2][4] = {
-   { 0x7f7fffff, 0x7f7fffff, 0x7f7fffff, 0x7f7fffff }, //1111
-   { 0x7fffffff, 0x7fffffff, 0x7fffffff, 0x7fffffff }, //1111
-};
-
 // Note: If modXYZW is true, then it adjusts XYZW for Single Scalar operations
 static void mVUupdateFlags(mV, const xmm& reg, const xmm& regT1in = xEmptyReg, const xmm& regT2in = xEmptyReg, bool modXYZW = 1) {
 	const x32&  mReg   = gprT1;
@@ -73,19 +67,6 @@ static void mVUupdateFlags(mV, const xmm& reg, const xmm& regT1in = xEmptyReg, c
 	if (mFLAG.doFlag) { SHIFT_XYZW(gprT2); }
 	xOR(mReg, gprT2);
 
-	if (sFLAG.doFlag) {
-		//Calculate overflow
-		xMOVAPS(regT1, regT2);
-		xAND.PS(regT1, ptr128[&sse4_compvals[1][0]]); // Remove sign flags (we don't care)
-		xPMIN.UD(regT1, ptr128[&sse4_compvals[0][0]]); // Get the minimum value, FLT_MAX = overflow
-		xCMPEQ.PS(regT1, ptr128[&sse4_compvals[0][0]]); // Compare if T1 == FLT_MAX
-		xMOVMSKPS(gprT2, regT1); // Grab sign bits  for equal results
-		xAND(gprT2, AND_XYZW);   // Grab "Is Zero" bits from the previous calculation
-		xTEST(gprT2, 0xF);
-		xForwardJump32 oJMP(Jcc_Zero);
-			xOR(sReg, 0x820000);
-		oJMP.SetTarget();
-	}
 	//-------------------------Write back flags------------------------------
 
 	if (mFLAG.doFlag) mVUallocMFLAGb(mVU, mReg, mFLAG.write); // Set Mac Flag
-- 
2.29.2

